File: src/main/java/resonantinduction/archaic/firebox/RenderHotPlate.java
package resonantinduction.archaic.firebox;

import net.minecraft.client.renderer.RenderBlocks;
import net.minecraft.tileentity.TileEntity;
import net.minecraftforge.common.ForgeDirection;
import resonantinduction.core.render.RenderItemOverlayTile;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class RenderHotPlate extends RenderItemOverlayTile
{
	private final RenderBlocks renderBlocks = new RenderBlocks();

	@Override
	public void renderTileEntityAt(TileEntity tileEntity, double x, double y, double z, float var8)
	{
		if (tileEntity instanceof TileHotPlate)
		{
			TileHotPlate tile = (TileHotPlate) tileEntity;
			renderTopOverlay(tileEntity, tile.getInventory().getContainedItems(), ForgeDirection.EAST, 2, 2, x, y - 0.8, z);
		}
	}
}

File: src/main/scala/resonantinduction/archaic/firebox/RenderHotPlate.scala
package resonantinduction.archaic.firebox

import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly
import net.minecraft.client.renderer.RenderBlocks
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import net.minecraftforge.common.util.ForgeDirection
import resonant.lib.render.RenderItemOverlayUtility

@SideOnly(Side.CLIENT) class RenderHotPlate extends TileEntitySpecialRenderer
{
    private final val renderBlocks: RenderBlocks = new RenderBlocks

    def renderTileEntityAt(tileEntity: TileEntity, x: Double, y: Double, z: Double, var8: Float)
    {
        if (tileEntity.isInstanceOf[TileHotPlate])
        {
            val tile: TileHotPlate = tileEntity.asInstanceOf[TileHotPlate]
            RenderItemOverlayUtility.renderTopOverlay(tileEntity, tile.getInventory.getContainedItems, ForgeDirection.EAST, 2, 2, x, y - 0.8, z, 1f)
        }
    }

}